deployer/src/runtime_manager.rs

```diff
@@ -0,0 +1,127 @@
+use std::{path::PathBuf, sync::Arc};
+
+use anyhow::Context;
+use shuttle_proto::runtime::{self, runtime_client::RuntimeClient, SubscribeLogsRequest};
+use tokio::{process, sync::Mutex};
+use tonic::transport::Channel;
+use tracing::{info, instrument, trace};
+
+use crate::deployment::deploy_layer;
+
+#[derive(Clone)]
+pub struct RuntimeManager {
+    legacy: Option<RuntimeClient<Channel>>,
+    legacy_process: Option<Arc<std::sync::Mutex<process::Child>>>,
+    next: Option<RuntimeClient<Channel>>,
+    next_process: Option<Arc<std::sync::Mutex<process::Child>>>,
+    artifacts_path: PathBuf,
+    provisioner_address: String,
+    log_sender: crossbeam_channel::Sender<deploy_layer::Log>,
+}
+
+impl RuntimeManager {
+    pub fn new(
+        artifacts_path: PathBuf,
+        provisioner_address: String,
+        log_sender: crossbeam_channel::Sender<deploy_layer::Log>,
+    ) -> Arc<Mutex<Self>> {
+        Arc::new(Mutex::new(Self {
+            legacy: None,
+            legacy_process: None,
+            next: None,
+            next_process: None,
+            artifacts_path,
+            provisioner_address,
+            log_sender,
+        }))
+    }
+
+    pub async fn get_runtime_client(
```

Comment:
Cool, you found a way! This'll be nice to have when we add more runtimes in the future as well.